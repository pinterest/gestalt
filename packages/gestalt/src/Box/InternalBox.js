
import { forwardRef, type Node, type AbstractComponent, type Element } from 'react';
import styles from '../Box.css';
import {blocklistProps} from '../Box.js';
import { buildStyles } from '../boxTransforms.js';
import { type As } from '../boxTypes.js';
import { type Indexable } from '../zIndex.js';

type OutputType = Element<As>;

/**
 * It's like Box, with all the conveniences, but also with custom styles allowed. A magic div with some utlities
 */
const InternalBoxWithForwardRef: AbstractComponent<Props, HTMLElement> = forwardRef<Props, HTMLElement>(
    function Box({ as, ...props }: Props, ref): OutputType {
      const { passthroughProps, propsStyles } = buildStyles<$Diff<Props, {| as?: As |}>>({
        baseStyles: styles.box,
        props,
        blocklistProps: disallowedProps,
      });
  
      const BoxElement: As = as ?? 'div';
  
      // And... magic!
      return <BoxElement {...passthroughProps} {...propsStyles} ref={ref} />;
    },
  );
  
  InternalBoxWithForwardRef.displayName = 'InternalBox';
  
  export default InternalBoxWithForwardRef;
  
