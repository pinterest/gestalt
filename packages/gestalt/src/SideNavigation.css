.ulItem {
  list-style: none;
  padding-inline-start: 0;
}

.liItem {
  margin-bottom: 0;
  margin-top: 0;
}

.borderExpandableSelected {
  border: 1px solid var(--color-black-cosmicore-900);
}

