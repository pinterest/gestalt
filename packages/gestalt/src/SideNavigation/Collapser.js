// @flow strict
import { type Node as ReactNode } from 'react';
import Box from '../Box';
import { useDefaultLabelContext } from '../contexts/DefaultLabelProvider';
import { useSideNavigation } from '../contexts/SideNavigationProvider';
import IconButton from '../IconButton';
import Sticky from '../Sticky';

export default function Collapser({ raised }: { raised: boolean }): ReactNode {
  const { collapsed, overlayPreview, onCollapse, setOverlayPreview } = useSideNavigation();
  const { accessibilityCollapseButtonLabel, accessibilityExpandButtonLabel } =
    useDefaultLabelContext('SideNavigation');

  return (
    <Sticky top={0}>
      <Box
        borderStyle={raised ? 'raisedTopShadow' : undefined}
        color={raised ? 'default' : undefined}
        display="flex"
        justifyContent={collapsed && !overlayPreview ? 'center' : 'end'}
        marginBottom={-2}
        padding={2}
      >
        <IconButton
          accessibilityLabel={
            collapsed ? accessibilityExpandButtonLabel : accessibilityCollapseButtonLabel
          }
          icon={collapsed ? 'arrow-double-forward' : 'arrow-double-back'}
          onClick={() => {
            onCollapse?.(!collapsed);

            if (collapsed) setOverlayPreview(false);
          }}
          size="xs"
        />
      </Box>
    </Sticky>
  );
}
