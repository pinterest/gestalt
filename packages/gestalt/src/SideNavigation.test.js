// @flow strict
import { create } from 'react-test-renderer';
import Box from './Box';
import SideNavigation from './SideNavigation';

describe('SideNavigation', () => {
  it('renders basic Item', () => {
    const tree = create(
      <SideNavigation accessibilityLabel="label">
        <SideNavigation.TopItem href="#" label="test" />
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Sections', () => {
    const tree = create(
      <SideNavigation accessibilityLabel="label">
        <SideNavigation.Section label="section">
          <SideNavigation.TopItem href="#" label="test" />
        </SideNavigation.Section>
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Icon + Badge/Notification + Counter + Border', () => {
    const tree = create(
      <SideNavigation accessibilityLabel="label" showBorder>
        <SideNavigation.Section label="section">
          <SideNavigation.TopItem
            badge={{ text: 'New', type: 'info' }}
            counter={{ number: '20', accessibilityLabel: 'You have 20 notifications' }}
            href="#"
            icon="bell"
            label="test"
          />
          <SideNavigation.TopItem
            counter={{ number: '20', accessibilityLabel: 'You have 20 notifications' }}
            href="#"
            icon="bell"
            label="test"
            notificationAccessibilityLabel="You have new messages"
          />
        </SideNavigation.Section>
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Header + Footer', () => {
    const tree = create(
      <SideNavigation
        accessibilityLabel="label"
        footer={<Box color="default" height={100} width="100%" />}
        header={<Box color="default" height={100} width="100%" />}
      >
        <SideNavigation.TopItem href="#" label="test" />
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders expandable nested directory', () => {
    const tree = create(
      <SideNavigation accessibilityLabel="Nested items example">
        <SideNavigation.TopItem
          href="#"
          icon="ads-stats"
          label="Reporting"
          onClick={({ event }) => event.preventDefault()}
        />
        <SideNavigation.TopItem
          href="#"
          icon="replace"
          label="Conversions"
          onClick={({ event }) => event.preventDefault()}
        />
        <SideNavigation.Section label="Audiences">
          <SideNavigation.TopItem
            href="#"
            icon="people"
            label="Thanksgiving"
            onClick={({ event }) => event.preventDefault()}
          />
          <SideNavigation.Group icon="people" label="Christmas">
            <SideNavigation.NestedItem
              href="#"
              label="Luxury Christmas"
              onClick={({ event }) => event.preventDefault()}
            />
            <SideNavigation.NestedGroup label="Classic Christmas">
              <SideNavigation.NestedItem
                href="#"
                label="West Coast"
                onClick={({ event }) => event.preventDefault()}
              />
              <SideNavigation.NestedItem
                href="#"
                label="East Coast"
                onClick={({ event }) => event.preventDefault()}
              />
            </SideNavigation.NestedGroup>
            <SideNavigation.NestedGroup label="Alternative Christmas">
              <SideNavigation.NestedItem
                href="#"
                label="West Coast"
                onClick={({ event }) => event.preventDefault()}
              />
              <SideNavigation.NestedItem
                href="#"
                label="East Coast"
                onClick={({ event }) => event.preventDefault()}
              />
            </SideNavigation.NestedGroup>
          </SideNavigation.Group>
          <SideNavigation.Group icon="people" label="Halloween">
            <SideNavigation.NestedItem
              href="#"
              label="East Coast"
              onClick={({ event }) => event.preventDefault()}
            />
            <SideNavigation.NestedItem
              href="#"
              label="West Coast"
              onClick={({ event }) => event.preventDefault()}
            />
          </SideNavigation.Group>
        </SideNavigation.Section>
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders static group', () => {
    const tree = create(
      <SideNavigation accessibilityLabel="Static items example">
        <SideNavigation.Group display="static" icon="people" label="Christmas">
          <SideNavigation.NestedItem
            href="#"
            label="Luxury Christmas"
            onClick={({ event }) => event.preventDefault()}
          />
          <SideNavigation.NestedGroup display="static" label="Classic Christmas">
            <SideNavigation.NestedItem
              href="#"
              label="West Coast"
              onClick={({ event }) => event.preventDefault()}
            />
            <SideNavigation.NestedItem
              href="#"
              label="East Coast"
              onClick={({ event }) => event.preventDefault()}
            />
          </SideNavigation.NestedGroup>
        </SideNavigation.Group>
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders expanded group', () => {
    const tree = create(
      <SideNavigation accessibilityLabel="Nested items example">
        <SideNavigation.Group display="expandable" expanded icon="people" label="Christmas">
          <SideNavigation.NestedItem
            href="#"
            label="Luxury Christmas"
            onClick={({ event }) => event.preventDefault()}
          />
          <SideNavigation.NestedGroup display="expandable" expanded label="Classic Christmas">
            <SideNavigation.NestedItem
              href="#"
              label="West Coast"
              onClick={({ event }) => event.preventDefault()}
            />
            <SideNavigation.NestedItem
              href="#"
              label="East Coast"
              onClick={({ event }) => event.preventDefault()}
            />
          </SideNavigation.NestedGroup>
        </SideNavigation.Group>
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
