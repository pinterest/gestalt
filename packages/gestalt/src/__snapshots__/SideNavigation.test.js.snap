// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation renders Header + Footer 1`] = `
<div>
  <nav
    aria-label="label"
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <div
          className="Flex columnGap4 xsDirectionColumn"
        >
          <div
            className="FlexItem"
          >
            <div
              className="box paddingX4"
            >
              <div
                className="box default"
                style={
                  Object {
                    "height": 100,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="FlexItem"
          >
            <hr
              className="divider"
            />
          </div>
        </div>
      </div>
      <div
        className="FlexItem"
      >
        <ul
          className="ulItem"
        >
          <li
            className="liItem"
          >
            <a
              className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              rel=""
              tabIndex={0}
              target={null}
            >
              <div
                className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                style={
                  Object {
                    "minHeight": 44,
                    "paddingInlineEnd": "16px",
                    "paddingInlineStart": "16px",
                  }
                }
              >
                <div
                  className="Flex rowGap2 xsDirectionRow"
                  style={
                    Object {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className="FlexItem flexGrow selfCenter"
                  >
                    <span
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      test
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </li>
        </ul>
      </div>
      <div
        className="FlexItem"
      >
        <div
          className="Flex columnGap4 xsDirectionColumn"
        >
          <div
            className="FlexItem"
          >
            <hr
              className="divider"
            />
          </div>
          <div
            className="FlexItem"
          >
            <div
              className="box paddingX4"
            >
              <div
                className="box default"
                style={
                  Object {
                    "height": 100,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`SideNavigation renders Icon + Badge/Notification + Counter + Border 1`] = `
<div
  className="borderRight"
>
  <nav
    aria-label="label"
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <ul
          className="ulItem"
        >
          <li
            className="liItem"
          >
            <div
              className="box marginBottom2 paddingX4 xsDisplayFlex"
              role="presentation"
            >
              <div
                className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                style={
                  Object {
                    "WebkitLineClamp": 2,
                  }
                }
                title="section"
              >
                section
              </div>
            </div>
            <ul
              className="ulItem"
            >
              <li
                className="liItem"
              >
                <a
                  className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  rel=""
                  tabIndex={0}
                  target={null}
                >
                  <div
                    className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                    style={
                      Object {
                        "minHeight": 44,
                        "paddingInlineEnd": "16px",
                        "paddingInlineStart": "16px",
                      }
                    }
                  >
                    <div
                      className="Flex rowGap2 xsDirectionRow"
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    >
                      <div
                        className="FlexItem selfCenter"
                      >
                        <div
                          aria-hidden={true}
                          className="box"
                        >
                          <svg
                            aria-hidden={true}
                            aria-label=""
                            className="icon defaultIcon"
                            height={16}
                            role="img"
                            viewBox="0 0 24 24"
                            width={16}
                          >
                            <path
                              d="test-file-stub"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        className="FlexItem flexGrow selfCenter"
                      >
                        <span
                          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                        >
                          test
                          <div
                            className="box marginStart1 xsDisplayInlineBlock"
                            style={
                              Object {
                                "height": "100%",
                              }
                            }
                          >
                            <div
                              className="box xsDisplayVisuallyHidden"
                            >
                              , 
                            </div>
                            <div
                              className="Badge middle infoBase"
                            >
                              New
                            </div>
                          </div>
                        </span>
                      </div>
                      <div
                        className="FlexItem flexNone selfCenter"
                      >
                        <div
                          className="box xsDisplayVisuallyHidden"
                        >
                          , 
                        </div>
                        <div
                          aria-label="You have 20 notifications"
                          className="box marginEndN2"
                          role="status"
                        >
                          <div
                            className="Text fontSize300 defaultText alignEnd breakWord fontWeightNormal"
                          >
                            20
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
              </li>
              <li
                className="liItem"
              >
                <a
                  className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  rel=""
                  tabIndex={0}
                  target={null}
                >
                  <div
                    className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                    style={
                      Object {
                        "minHeight": 44,
                        "paddingInlineEnd": "16px",
                        "paddingInlineStart": "16px",
                      }
                    }
                  >
                    <div
                      className="Flex rowGap2 xsDirectionRow"
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    >
                      <div
                        className="FlexItem selfCenter"
                      >
                        <div
                          aria-hidden={true}
                          className="box"
                        >
                          <svg
                            aria-hidden={true}
                            aria-label=""
                            className="icon defaultIcon"
                            height={16}
                            role="img"
                            viewBox="0 0 24 24"
                            width={16}
                          >
                            <path
                              d="test-file-stub"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        className="FlexItem flexGrow selfCenter"
                      >
                        <span
                          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                        >
                          test
                          <div
                            className="box marginStart1 xsDisplayInlineBlock"
                            style={
                              Object {
                                "height": "100%",
                              }
                            }
                          >
                            <div
                              className="box xsDisplayVisuallyHidden"
                            >
                              , 
                            </div>
                            <div
                              aria-label="You have new messages"
                              className="box circle primary"
                              role="status"
                              style={
                                Object {
                                  "height": 8,
                                  "width": 8,
                                }
                              }
                            />
                          </div>
                        </span>
                      </div>
                      <div
                        className="FlexItem flexNone selfCenter"
                      >
                        <div
                          className="box xsDisplayVisuallyHidden"
                        >
                          , 
                        </div>
                        <div
                          aria-label="You have 20 notifications"
                          className="box marginEndN2"
                          role="status"
                        >
                          <div
                            className="Text fontSize300 defaultText alignEnd breakWord fontWeightNormal"
                          >
                            20
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`SideNavigation renders Sections 1`] = `
<div>
  <nav
    aria-label="label"
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <ul
          className="ulItem"
        >
          <li
            className="liItem"
          >
            <div
              className="box marginBottom2 paddingX4 xsDisplayFlex"
              role="presentation"
            >
              <div
                className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                style={
                  Object {
                    "WebkitLineClamp": 2,
                  }
                }
                title="section"
              >
                section
              </div>
            </div>
            <ul
              className="ulItem"
            >
              <li
                className="liItem"
              >
                <a
                  className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  rel=""
                  tabIndex={0}
                  target={null}
                >
                  <div
                    className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                    style={
                      Object {
                        "minHeight": 44,
                        "paddingInlineEnd": "16px",
                        "paddingInlineStart": "16px",
                      }
                    }
                  >
                    <div
                      className="Flex rowGap2 xsDirectionRow"
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    >
                      <div
                        className="FlexItem flexGrow selfCenter"
                      >
                        <span
                          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                        >
                          test
                        </span>
                      </div>
                    </div>
                  </div>
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`SideNavigation renders basic Item 1`] = `
<div>
  <nav
    aria-label="label"
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <ul
          className="ulItem"
        >
          <li
            className="liItem"
          >
            <a
              className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              rel=""
              tabIndex={0}
              target={null}
            >
              <div
                className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                style={
                  Object {
                    "minHeight": 44,
                    "paddingInlineEnd": "16px",
                    "paddingInlineStart": "16px",
                  }
                }
              >
                <div
                  className="Flex rowGap2 xsDirectionRow"
                  style={
                    Object {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className="FlexItem flexGrow selfCenter"
                  >
                    <span
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      test
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`SideNavigation renders nested directory 1`] = `
<div>
  <nav
    aria-label="Nested items example"
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <ul
          className="ulItem"
        >
          <li
            className="liItem"
          >
            <a
              className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              rel=""
              tabIndex={0}
              target={null}
            >
              <div
                className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                style={
                  Object {
                    "minHeight": 44,
                    "paddingInlineEnd": "16px",
                    "paddingInlineStart": "16px",
                  }
                }
              >
                <div
                  className="Flex rowGap2 xsDirectionRow"
                  style={
                    Object {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className="FlexItem selfCenter"
                  >
                    <div
                      aria-hidden={true}
                      className="box"
                    >
                      <svg
                        aria-hidden={true}
                        aria-label=""
                        className="rtlSupport icon defaultIcon"
                        height={16}
                        role="img"
                        viewBox="0 0 24 24"
                        width={16}
                      >
                        <path
                          d="test-file-stub"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    className="FlexItem flexGrow selfCenter"
                  >
                    <span
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      Reporting
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </li>
          <li
            className="liItem"
          >
            <a
              className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              rel=""
              tabIndex={0}
              target={null}
            >
              <div
                className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                style={
                  Object {
                    "minHeight": 44,
                    "paddingInlineEnd": "16px",
                    "paddingInlineStart": "16px",
                  }
                }
              >
                <div
                  className="Flex rowGap2 xsDirectionRow"
                  style={
                    Object {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className="FlexItem selfCenter"
                  >
                    <div
                      aria-hidden={true}
                      className="box"
                    >
                      <svg
                        aria-hidden={true}
                        aria-label=""
                        className="icon defaultIcon"
                        height={16}
                        role="img"
                        viewBox="0 0 24 24"
                        width={16}
                      >
                        <path
                          d="test-file-stub"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    className="FlexItem flexGrow selfCenter"
                  >
                    <span
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      Conversions
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </li>
          <li
            className="liItem marginTop6"
          >
            <div
              className="box marginBottom2 paddingX4 xsDisplayFlex"
              role="presentation"
            >
              <div
                className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                style={
                  Object {
                    "WebkitLineClamp": 2,
                  }
                }
                title="Audiences"
              >
                Audiences
              </div>
            </div>
            <ul
              className="ulItem"
            >
              <li
                className="liItem"
              >
                <a
                  className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  rel=""
                  tabIndex={0}
                  target={null}
                >
                  <div
                    className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                    style={
                      Object {
                        "minHeight": 44,
                        "paddingInlineEnd": "16px",
                        "paddingInlineStart": "16px",
                      }
                    }
                  >
                    <div
                      className="Flex rowGap2 xsDirectionRow"
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    >
                      <div
                        className="FlexItem selfCenter"
                      >
                        <div
                          aria-hidden={true}
                          className="box"
                        >
                          <svg
                            aria-hidden={true}
                            aria-label=""
                            className="icon defaultIcon"
                            height={16}
                            role="img"
                            viewBox="0 0 24 24"
                            width={16}
                          >
                            <path
                              d="test-file-stub"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        className="FlexItem flexGrow selfCenter"
                      >
                        <span
                          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                        >
                          Thanksgiving
                        </span>
                      </div>
                    </div>
                  </div>
                </a>
              </li>
              <li
                className="liItem"
              >
                <div
                  aria-controls=":r8:"
                  aria-disabled={false}
                  aria-expanded={false}
                  className="hideOutline tapTransition rounding2 accessibilityOutline fullWidth pointer"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                    style={
                      Object {
                        "minHeight": 44,
                        "paddingInlineEnd": "16px",
                        "paddingInlineStart": "16px",
                      }
                    }
                  >
                    <div
                      className="Flex rowGap2 xsDirectionRow"
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    >
                      <div
                        className="FlexItem selfCenter"
                      >
                        <div
                          aria-hidden={true}
                          className="box"
                        >
                          <svg
                            aria-hidden={true}
                            aria-label=""
                            className="icon defaultIcon"
                            height={16}
                            role="img"
                            viewBox="0 0 24 24"
                            width={16}
                          >
                            <path
                              d="test-file-stub"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        className="FlexItem flexGrow selfCenter"
                      >
                        <span
                          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                        >
                          Christmas
                        </span>
                      </div>
                      <div
                        className="FlexItem flexNone selfCenter"
                      >
                        <div
                          aria-hidden={true}
                          className="box circle marginEndN2 marginStart2"
                          tabIndex={-1}
                        >
                          <svg
                            aria-hidden={true}
                            aria-label=""
                            className="icon subtleIcon iconBlock"
                            height={12}
                            role="img"
                            viewBox="0 0 24 24"
                            width={12}
                          >
                            <path
                              d="test-file-stub"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </li>
              <li
                className="liItem"
              >
                <div
                  className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                  style={
                    Object {
                      "minHeight": 44,
                      "paddingInlineEnd": "16px",
                      "paddingInlineStart": "16px",
                    }
                  }
                >
                  <div
                    className="Flex rowGap2 xsDirectionRow"
                    style={
                      Object {
                        "height": "100%",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      className="FlexItem selfCenter"
                    >
                      <div
                        aria-hidden={true}
                        className="box"
                      >
                        <svg
                          aria-hidden={true}
                          aria-label=""
                          className="icon defaultIcon"
                          height={16}
                          role="img"
                          viewBox="0 0 24 24"
                          width={16}
                        >
                          <path
                            d="test-file-stub"
                          />
                        </svg>
                      </div>
                    </div>
                    <div
                      className="FlexItem flexGrow selfCenter"
                    >
                      <span
                        className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                      >
                        Halloween
                      </span>
                    </div>
                  </div>
                </div>
                <ul
                  className="ulItem"
                  id=":r9:"
                >
                  <li
                    className="liItem"
                  >
                    <a
                      className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                      href="#"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchCancel={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      rel=""
                      tabIndex={0}
                      target={null}
                    >
                      <div
                        className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                        style={
                          Object {
                            "minHeight": 44,
                            "paddingInlineEnd": "16px",
                            "paddingInlineStart": "48px",
                          }
                        }
                      >
                        <div
                          className="Flex rowGap2 xsDirectionRow"
                          style={
                            Object {
                              "height": "100%",
                              "width": "100%",
                            }
                          }
                        >
                          <div
                            className="FlexItem flexGrow selfCenter"
                          >
                            <span
                              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                            >
                              East Coast
                            </span>
                          </div>
                        </div>
                      </div>
                    </a>
                  </li>
                  <li
                    className="liItem"
                  >
                    <a
                      className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                      href="#"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchCancel={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      rel=""
                      tabIndex={0}
                      target={null}
                    >
                      <div
                        className="box itemsCenter paddingY2 rounding2 xsDisplayFlex"
                        style={
                          Object {
                            "minHeight": 44,
                            "paddingInlineEnd": "16px",
                            "paddingInlineStart": "48px",
                          }
                        }
                      >
                        <div
                          className="Flex rowGap2 xsDirectionRow"
                          style={
                            Object {
                              "height": "100%",
                              "width": "100%",
                            }
                          }
                        >
                          <div
                            className="FlexItem flexGrow selfCenter"
                          >
                            <span
                              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                            >
                              West Coast
                            </span>
                          </div>
                        </div>
                      </div>
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
